shipyard_utils/files.py

```diff
@@ -102,6 +104,107 @@ def determine_destination_full_path(
+
+    if compression == 'zip':
+        with ZipFile(compressed_file_name, 'w') as zip:
+            for file in file_paths:
+                file = clean_folder_name(file.replace(os.getcwd(), ''))
+                zip.write(file)
+                print(
+                    f'Successfully compressed {file} into {compressed_file_name}')
+
+    if compression == 'tar.bz2':
+        with tarfile.open(compressed_file_name, 'w:bz2') as tar:
+            for file in file_paths:
+                file = clean_folder_name(file.replace(os.getcwd(), ''))
+                tar.add(file)
+                print(
+                    f'Successfully compressed {file} into {compressed_file_name}')
+
+    if compression == 'tar':
+        with tarfile.open(compressed_file_name, 'w') as tar:
+            for file in file_paths:
+                file = clean_folder_name(file.replace(os.getcwd(), ''))
+                tar.add(file)
+                print(
+                    f'Successfully compressed {file} into {compressed_file_name}')
+
+    if compression == 'tar.gz':
+        with tarfile.open(compressed_file_name, 'w:gz') as tar:
+            for file in file_paths:
+                file = clean_folder_name(file.replace(os.getcwd(), ''))
+                tar.add(file)
+                print(
+                    f'Successfully compressed {file} into {compressed_file_name}')
+
+    return compressed_file_name
+
+
+def decompress_file(source_full_path, destination_full_path, compression):
+    """
+    Decompress a given file, using the specified compression method.
+    """
+
+    if compression == 'zip':
+        with ZipFile(source_full_path, 'r') as zip:
+            zip.extractall(destination_full_path)
+        print(
+            f'Successfully extracted files from {source_full_path} to {destination_full_path}')
+
+    if compression == 'tar.bz2':
+        file = tarfile.open(source_full_path, 'r:bz2')
+        file.extractall(path=destination_full_path)
+        print(
+            f'Successfully extracted files from {source_full_path} to {destination_full_path}')
+
+    if compression == 'tar':
+        file = tarfile.open(source_full_path, 'r')
+        file.extractall(path=destination_full_path)
+        print(
+            f'Successfully extracted files from {source_full_path} to {destination_full_path}')
+
+    if compression == 'tar.gz':
+        file = tarfile.open(source_full_path, 'r:gz')
+        file.extractall(path=destination_full_path)
+        print(
+            f'Successfully extracted files from {source_full_path} to {destination_full_path}')
+
+
+def is_file_too_large(file_path, max_size_bytes=10000000):
+    """
+    Determine if the file is too large for a specified limit.
+    Used to conditionally compress a file.
+    """
+    if os.stat(file_path).st_size >= max_size_bytes:
+        return True
+    else:
+        return False
+
+
+def are_files_too_large(file_paths, max_size_bytes=10000000):
```

Comment:
Create a constant `TEN_MB` or the like for the max_size_bytes value. 
I know you're thinking of email for `is_file_too_large`, but the function name doesn't necessarily mean email. Is it too large for s3, no; email, depends.

Reply:
@ericelsken I'm only including a default for convenience. Since any script could use any value (not just email), it doesn't make sense to me to store a constant in the `utils`.

I'd rather have no default than include a constant in the `utils`. Is that fine?

Reply:
Yep [](edited-with-Axolo)